controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,39 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment)
+                        .map(x -> new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return reduce(oldRanges).equals(reduce(newRanges));
```

Comment:
I'm never that the implementation of `equals` is doing the right thing out of the box. Could you make sure that this is the case?